runtime/builtin_types.go

```diff
@@ -637,3 +627,31 @@ func numberToBase(prefix string, base int, o *Object) string {
 	}
 	return prefix + s
 }
+
+// quantifyIterable implements the builtins "all" and "any".
+// It takes a condition function, "cond", that determines when
+// the iteration should stop.  If the iteration stops before
+// all the items are seen, then "condTrue" is returned. Otherwise,
+// "condFalse" is returned. For more details see:
+// https://docs.python.org/2/library/functions.html#all
+func quantifyIterable(f *Frame, o *Object, cond func(bool) bool, condTrue *Object, condFalse *Object) (*Object, *BaseException) {
```

Comment:
How about something like this:

```
func seqFindFirst(f *Frame, iterable *Object, pred func(*Object) bool) (bool, *BaseException)
```

Then you can define predicates `seqFindFirstTrue()` and `seqFindFirstFalse()` to be used by `any` and `all`. `seqFindFirst()` can then be reused by `[grumpy.Contains()](https://github.com/google/grumpy/blob/master/runtime/core.go#L71)`.

What do you think?